python/py_vapid/__init__.py

```diff
@@ -100,54 +98,69 @@ def from_file(cls, private_key_file=None):
         """
         if not os.path.isfile(private_key_file):
             vapid = cls()
+            vapid.generate_keys()
             vapid.save_key(private_key_file)
             return vapid
         private_key = open(private_key_file, 'r').read()
-        vapid = None
         try:
-            if "BEGIN EC" in private_key:
-                vapid = cls.from_pem(private_key)
+            if "-----BEGIN" in private_key:
+                vapid = cls.from_pem(private_key.encode('utf8'))
             else:
-                vapid = cls.from_der(private_key)
+                vapid = cls.from_der(private_key.encode('utf8'))
+            return vapid
         except Exception as exc:
             logging.error("Could not open private key file: %s", repr(exc))
             raise VapidException(exc)
-        return vapid
 
     @property
     def private_key(self):
         """The VAPID private ECDSA key"""
         if not self._private_key:
-            raise VapidException(
-                "No private key defined. Please import or generate a key.")
+            raise VapidException("No private key. Call generate_keys()")
         return self._private_key
 
     @private_key.setter
     def private_key(self, value):
         """Set the VAPID private ECDSA key
 
         :param value: the byte array containing the private ECDSA key data
-        :type value: bytes
+        :type value: ec.EllipticCurvePrivateKey
 
         """
         self._private_key = value
-        self._public_key = None
+        if value:
+            self._public_key = self.private_key.public_key()
 
     @property
     def public_key(self):
         """The VAPID public ECDSA key
 
         The public key is currently read only. Set it via the `.private_key`
-        method.
+        method. This will autogenerate a public and private key if no value
```

Comment:
Yep, I realized that doing the "optimization" could break some installs, forgot to clean this one up.